draft-bishop-httpbis-http2-additional-certs.md

```diff
@@ -322,53 +321,49 @@ Clients and servers that will accept requests for HTTP-layer certificate
-indicating that the peer does not support HTTP-layer certificate authentication.
-If a peer does support HTTP-layer certificate authentication, the value
-is 1.
+The initial value for the `SETTINGS_HTTP_CERT_AUTH` setting is 0, indicating
+that the peer does not support HTTP-layer certificate authentication. If a peer
+does support HTTP-layer certificate authentication, the value is 1.
 
 ## Making certificates or requests available {#cert-available}
 
-When a peer has advertised support for HTTP-layer certificates as in
-{{setting}}, either party can supply additional certificates into the
-connection at any time. These certificates then become available for the
-peer to consider when deciding whether a connection is suitable to
-transport a particular request.
+When both peers have advertised support for HTTP-layer certificates as in
+{{setting}}, either party can supply additional certificates into the connection
+at any time. This means that clients or servers which predict a certificate will
+be required could pre-supply the certificate before being asked. These
+certificates are available for reference by future `USE_CERTIFICATE` frames.
 
-Available certificates which have the `AUTOMATIC_USE` flag set MAY be
-used by the recipient without further notice. This means that clients or
-servers which predict a certificate will be required could pre-supply
-the certificate without being asked. Regardless of whether
-`AUTOMATIC_USE` is set, these certificates are available for reference
-by future `USE_CERTIFICATE` frames.
+Certificates supplied by servers can be considered by clients without further
+action by the server. A server MUST NOT send certificates for origins which it
+is not prepared to service on the current connection.
 
 ~~~
 Client                                      Server
-   <-------- (stream 0) CERTIFICATE (AU flag) --
+   <------------------ (stream 0) CERTIFICATE --
    ...
    -- (stream N) GET /from-new-origin --------->
    <----------------------- (stream N) 200 OK --
-
 ~~~
 {: #ex-http2-server-proactive title="Proactive Server Certificate"}
 
 ~~~
 Client                                      Server
-   -- (stream 0) CERTIFICATE (AU flag) -------->
+   -- (stream 0) CERTIFICATE ------------------>
    -- (streams 1,3) GET /protected ------------>
+   -- (streams 1,3) USE_CERTIFICATE ----------->
    <-------------------- (streams 1,3) 200 OK --
-
 ~~~
 {: #ex-http2-client-proactive title="Proactive Client Certificate"}
 
 Likewise, either party can supply a `CERTIFICATE_REQUEST` that outlines
-parameters of a certificate they might request in the future.  It
-is important to note that this does not currently request such a
-certificate, but makes the contents of the request available for
-reference by a future `CERTIFICATE_NEEDED` frame.
+parameters of a certificate they might request in the future.  Upon
+receipt of a `CERTIFICATE_REQUEST`, the other party SHOULD provide a
+corresponding certificate.
```

Comment:
I think that we should provide an explicit indication that a certificate will not be provided.  Certificate authentication is often something that you can't put a timer around.  User prompts are involved, which can take an indeterminate time to resolve.  That means that servers in particular will get stuck if the client intends to not provide a response.  An explicit indication that the certificate won't be forthcoming would be good.  TLS does that by including an empty Certificate message.